File: backend/src/main/java/patient/scheduling/system/api/domain/entity/Schedule.java
package patient.scheduling.system.api.domain.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import patient.scheduling.system.api.domain.enums.StatusENUM;

import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonFormat(pattern = "dd/MM/yyyy HH:mm")
    @Column(nullable = false)
    private LocalDateTime dateTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusENUM status;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Medic medic;

    @ManyToOne
    private Patient patient;
}

File: frontend/src/app/schedules/domain/Schedule.ts
import { Medic } from "src/app/medics/domain/medic";
import { Patient } from "src/app/patients/domain/patient";

export interface Schedule {
  id: Number,
  dateTime: String,
  status: String,
  medic: Medic | null,
  patient: Patient | null
}
